File: LeetCode/DataStruct/JS/Leetcode921-minimum-add-to-make-parentheses-valid.js
var minAddToMakeValid = function (s) {
  let [l, r] = [0, 0];
  for (const c of s) {
    if (c == '(') l++;
    else if (c == ')') {
      if (!l) r++;
      else l--;
    }
  }
  return l + r;
};

File: LeetCode/DataStruct/Golang/Leetcode921-minimum-add-to-make-parentheses-valid.go
package sbgo

func minAddToMakeValid(s string) int {
	l, r := 0, 0
	for _, c := range s {
		if c == '(' {
			l++
		} else if c == ')' {
			if l == 0 {
				r++
			} else {
				l--
			}
		}
	}
	return l + r
}
